src/repr/scalar/datetime.rs

```diff
@@ -1921,15 +1921,7 @@ mod test {
                 "0-0 0",
                 Year,
                 1,
-                "Invalid syntax at offset 1: provided Space but expected Dash",
-            ),
-            // If you have Nanos, you cannot convert to Num.
-            (
-                "1 2.3",
-                "0 0.0YEAR",
-                Year,
-                1,
-                "Invalid syntax at offset 4: provided Nanos(300000000) but expected TimeUnit(Year)",
+                "Invalid syntax at offset 1: provided Delim but expected Dash",
```

Comment:
maybe in a future commit we could change `Delim` -> `Delim(char)` so that this would look like `provided Delim(' ') but expected Dash`, although we are including the offset so it's probably fine.

Reply:
That's a good call. I've been being bitten a lot by how fussy using embedded values in enums are when you're doing a lot of matching, so I felt averse at first, but it would dramatically improve the legibility of the errors--you're right.